Scripts/main.js

```diff
@@ -0,0 +1,181 @@
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.maxLength = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = ruleValue.length <= parametr ? true : false;
+        var message = ruleName + "'s(" + ruleValue + ')' + " length" + '('
+            + ruleValue.length + ')' + " must be less than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.minLength = function (value) {
+    var parametr = value;
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = ruleValue.length >= parametr ? true : false;
+        var message = ruleName + "'s(" + ruleValue + ')' + " length" + '('
+            + ruleValue.length + ')' + " must be greater than " + parametr;
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.isInt = function () {
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var isValid = !isNaN(parseFloat(ruleValue) && isFinite(ruleValue));
+        var message = ruleName + '(' + ruleValue + ')' + " must be a number";
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+Rule.prototype.isEmail = function () {
+    this.functionArray.push(function (ruleValue, ruleName) {
+        var emailReg = /^[a-z][a-zA-Z0-9_.]*(\.[a-zA-Z][a-zA-Z0-9_.]*)?@[a-z][a-zA-Z0-9]*\.[a-z]+(\.[a-z]+)?$/;
+        var isValid = emailReg.test(ruleValue);
+        var message = ruleName + '(' + ruleValue + ')' + " must be an email";
+        var validationResult = {
+            errorMessage: isValid ? null : message,
+            isValid: isValid
+        }
+        return validationResult;
+    });
+    return this;
+}
+
+var Validator = function () {
+
+}
+
+Validator.validate = function (obj, rules) {
+    var properties = Object.getOwnPropertyNames(obj);
+    var propToRule = Object.getOwnPropertyNames(rules);
+    var errorsArray = [];
+    for (var i = 0; i < properties.length; i++) {
+        rules[propToRule[i]].value = obj[properties[i]];
+    }
+    for (var i = 0; i < properties.length; i++) {
+        var rule = rules[properties[i]];
+        for (var j = 0; j < rule.functionArray.length; j++) {
+            var result = rule.functionArray[j].call(rule, rules[propToRule[i]].value, propToRule[i]);
+            if (!result.isValid) {
+                errorsArray.push(result.errorMessage)
+            }
+        }
+    }
+    if (errorsArray.length === 0) {
+        return new Promise((resolve, reject) => {
```

Comment:
Need to know difference between arrow functions and anonymous functions. Please read.